src/main/steps/common/switch-to-sole-content.utils.ts

```diff
@@ -0,0 +1,104 @@
+import config from 'config';
+import dayjs from 'dayjs';
+
+import { CaseWithId } from '../../app/case/case';
+import { State, YesOrNo } from '../../app/case/definition';
+
+import { hasApplicantAppliedForFinalOrderFirst } from './content.utils';
+
+interface SwitchToSoleFinalOrderStatus {
+  isWithinSwitchToSoleFinalOrderIntentionNotificationPeriod: boolean;
+  hasSwitchToSoleFinalOrderIntentionNotificationPeriodExpired: boolean;
+  isIntendingAndAbleToSwitchToSoleFinalOrder: boolean;
+}
+
+export const doesApplicantIntendToSwitchToSoleFinalOrder = (
+  userCase: Partial<CaseWithId>,
+  isApplicant2: boolean
+): boolean => {
+  return isApplicant2
+    ? userCase.doesApplicant2IntendToSwitchToSole === YesOrNo.YES
+    : userCase.doesApplicant1IntendToSwitchToSole === YesOrNo.YES;
+};
+
+const dateApplicantDeclaredIntentionToSwitchToSoleFinalOrder = (
+  userCase: Partial<CaseWithId>,
+  isApplicant2: boolean
+): string | undefined => {
+  return isApplicant2
+    ? userCase.dateApplicant2DeclaredIntentionToSwitchToSoleFo
+    : userCase.dateApplicant1DeclaredIntentionToSwitchToSoleFo;
+};
+
+const hasApplicantDeclaredIntentionToSwitchToSoleFinalOrder = (
+  userCase: Partial<CaseWithId>,
+  isApplicant2: boolean
+): boolean => {
+  return (
+    hasApplicantAppliedForFinalOrderFirst(userCase, isApplicant2) &&
+    doesApplicantIntendToSwitchToSoleFinalOrder(userCase, isApplicant2) &&
+    dayjs().isAfter(dateApplicantDeclaredIntentionToSwitchToSoleFinalOrder(userCase, isApplicant2))
```

Comment:
Do we need both?

Reply:
Adding too many conditions that check the same sort of thing might be a bit confusing to people in the future

Reply:
It was setting up a check elsewhere which should have a start and end date.  However this is only required in one other location and not each time this method is called.  I have refactored this.

Reply:
Can we move it to isWithinSwitchToSoleFinalOrderIntentionNotificationPeriod method